src/common/connection/DuckDbConnection.ts

```diff
@@ -0,0 +1,65 @@
+import type { RootConfig } from "$common/config/RootConfig";
+import { DATABASE_POLLING_INTERVAL } from "$common/constants";
+import type { DataModelerService } from "$common/data-modeler-service/DataModelerService";
+import type { DataModelerStateService } from "$common/data-modeler-state-service/DataModelerStateService";
+import { EntityType, StateType } from "$common/data-modeler-state-service/entity-state-service/EntityStateService";
+import type { DuckDBClient } from "$common/database-service/DuckDBClient";
+import { DataConnection } from "./DataConnection";
+import type {
+    PersistentTableEntity
+} from "$common/data-modeler-state-service/entity-state-service/PersistentTableEntityService";
+
+/**
+ * Connects to an existing duck db.
+ * Adds all existing table into the table states.
+ * Periodically syncs
+ */
+export class DuckDbConnection extends DataConnection {
+    private syncTimer: NodeJS.Timer;
+
+    public constructor(
+        protected readonly config: RootConfig,
+        protected readonly dataModelerService: DataModelerService,
+        protected readonly dataModelerStateService: DataModelerStateService,
+        protected readonly duckDbClient: DuckDBClient,
+    ) {
+        super(config, dataModelerService, dataModelerStateService);
+    }
+
+    public async init(): Promise<void> {
+        await this.sync();
+
+        this.syncTimer = setInterval(() => {
+            this.sync();
+        }, DATABASE_POLLING_INTERVAL);
+    }
+
+    public async sync(): Promise<void> {
+        const tables = await this.duckDbClient.execute("SHOW TABLES");
```

Comment:
I am seeing that this will actually pick up temporary views and temporary tables in the list of available tables. We're currently using views and might end up using temporary tables down the line, so we should switch this to querying the [information schema](https://duckdb.org/docs/sql/information_schema) directly like this:

```sql
SELECT * FROM information_schema.tables
WHERE table_type NOT ILIKE '%TEMPORARY' AND table_type NOT ILIKE '%VIEW'
```

Otherwise the interface mistakes temporary tables + views in the table set.

It would make sense to add a test for this I think.

Reply:
Sure. Lemme fix this.